packages/graphql/tests/tck/tck-test-files/union.test.ts

```diff
@@ -54,7 +54,114 @@ describe("Cypher Union", () => {
         });
     });
 
-    test("Read Unions", async () => {
+    test("Read Unions simple", async () => {
+        const query = gql`
+            {
+                movies {
+                    search {
+                        ... on Movie {
+                            title
+                        }
+                        ... on Genre {
+                            name
+                        }
+                    }
+                }
+            }
+        `;
+
+        const req = createJwtRequest("secret", { jwtAllowedNamesExample: ["Horror"] });
+        const result = await translateQuery(neoSchema, query, {
+            req,
+        });
+
+        expect(formatCypher(result.cypher)).toMatchInlineSnapshot(`
+            "MATCH (this:\`Movie\`)
+            CALL {
+                WITH this
+                CALL {
+                    WITH this
+                    MATCH (this)-[thisthis0:SEARCH]->(this_search_0:\`Genre\`)
+                    WHERE apoc.util.validatePredicate(NOT ((this_search_0.name IS NOT NULL AND this_search_0.name = $thisparam0)), \\"@neo4j/graphql/FORBIDDEN\\", [0])
+                    WITH this_search_0  { __resolveType: \\"Genre\\",  .name } AS this_search_0
+                    RETURN collect(this_search_0) AS this_search_0
+                }
+                CALL {
+                    WITH this
+                    MATCH (this)-[thisthis1:SEARCH]->(this_search_1:\`Movie\`)
+                    WITH this_search_1  { __resolveType: \\"Movie\\",  .title } AS this_search_1
+                    RETURN collect(this_search_1) AS this_search_1
+                }
+                WITH this_search_0 + this_search_1 AS thisvar2
```

Comment:
For this whole union piece, I have to ask why the approach as per below wasn't taken? It feels like Cypher subquery unions are the perfect solution here (and what we use elsewhere). This would also mean you can fully get rid of the unique (0, 1, etc.) variables in each subquery, which I would guess massively simplifies the solution.

```suggestion
                CALL {
                    WITH this
                    MATCH (this)-[thisthis0:SEARCH]->(this_search_0:\`Genre\`)
                    WHERE apoc.util.validatePredicate(NOT ((this_search_0.name IS NOT NULL AND this_search_0.name = $thisparam0)), \\"@neo4j/graphql/FORBIDDEN\\", [0])
                    WITH this_search_0  { __resolveType: \\"Genre\\",  .name } AS this_search_0
                    RETURN collect(this_search_0) AS this_search
                UNION
                    WITH this
                    MATCH (this)-[thisthis1:SEARCH]->(this_search_1:\`Movie\`)
                    WITH this_search_1  { __resolveType: \\"Movie\\",  .title } AS this_search_1
                    RETURN collect(this_search_1) AS this_search
                }
```

Reply:
That's a good point, I already considered doing this, but I found a few roadblocks when adding sorting and discarded the idea. However, after doing a bit more manual testing it is indeed doable (although, you still need to add a second CALL for the sorting)

All tests seem to be working after this change, so I'll update this PR for using unions

Reply:
Done, please, review latest changes